lib/csv_format_guesser.rb

```diff
@@ -37,8 +38,8 @@ def guess_encoding
   end
 
   def try_encoding_with_fallback!
-    File.open(@path, "r", encoding: @encoding) do |f|
-      f.read
+    File.foreach(@path, encoding: @encoding).first(MAX_LINES).each_with_object(true) do |line, r|
+      r &&= line.valid_encoding?
```

Comment:
`each_with_object` doesn't use the return value of the block, it always returns the memo value passed to the method at the beginning (in your case `true`) so this is guaranteed to always return `true`

Reply:
oh true, thanks for pointing that out. run it with broken files in console, but not after porting it to this repo... and writing it a bit shorter 😅